### PR TITLE
Add release job

### PR DESCRIPTION
**Pull request by yordis**
## Context

Allows releasing the package from CI.

## Requirements

`HEX_API_KEY` GitHub Secret to be set up, https://github.com/straw-hat-team/github-actions-workflows/tree/master/elixir/publish#add-a-hex-api-key-to-github-secrets

## How does it work?

When a new release is published, it will try to release a new version to Hex.

### CODE REVIEW

**Comment by tony612** [comment]
I can help with adding the key and env. But I may need @whatyouhide or someone else's reviewing given I'm not maintaining Protobuf related projects for a long time.

**Comment by yordis** [comment]
@tony612 adding the `HEX_API_KEY` will allow them to cut a release by creating a GitHub Release, or it could be modified to use tag creation instead.

So it will help for sure.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Hexpm Release
```
Comment (whatyouhide) [inline .github/workflows/release.yml:1]:
```suggestion
name: Release on Hex
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Elixir
+      uses: erlef/setup-elixir@v1
```
Comment (whatyouhide) [inline .github/workflows/release.yml:15]:
```suggestion
      uses: erlef/setup-beam@v1
```

**Pull request merged**